labgrid/resource/remote.py

```diff
@@ -113,12 +113,20 @@ class RemoteUSBResource(NetworkResource, ManagedResource):
 
 @target_factory.reg_resource
 @attr.s(eq=False)
-class NetworkAndroidFastboot(RemoteUSBResource):
+class NetworkAndroidUsbFastboot(RemoteUSBResource):
     def __attrs_post_init__(self):
         self.timeout = 10.0
         super().__attrs_post_init__()
 
 
+@target_factory.reg_resource
+@attr.s(eq=False)
+class NetworkAndroidNetFastboot(NetworkResource):
+    address = attr.ib(validator=attr.validators.instance_of(str))
+    port = attr.ib(default=5554, validator=attr.validators.instance_of(int))
+    protocol = attr.ib(default="udp", validator=attr.validators.instance_of(str))
+
+
```

Comment:
This seems identical to the AndroidNetFastboot resource, so it shouldn't be needed?

Reply:
Yes it seems, because the attributes are identical. But the inherited type is different. Our understanding is that we need two separate ressources. One local and one remote resource. In case of the remote resource we need  to inherit from `NetworkResource` to get the ssh command prefix. Is this correct?

Reply:
Ah, I had expected that with network fastboot, you'd just use SSH port forwarding. But now I understand that for UDP, that can't work and that's why you still execute fastboot on the exporter.

To clear up the naming confusion, we've started to name new exported resources Remote* (see RemoteNetworkInterface). So for this case, I'd use RemoteAndroidNetFastboot.

In the medium term, if we get SSH SOCKS5 tunnels, we could run network fastboot over that tunnel, but that can be added later.